pydra/mark/tests/test_shell.py

```diff
@@ -0,0 +1,467 @@
+    assert b.inputs.hidden
+    assert b.cmdline == f"ls -a {tmpdir}"
+
+    result = b()
+    assert result.output.entries == [".", "..", ".hidden"]
+
+
+def test_shell_inputs_outputs_bases_static(tmpdir):
+    @shell_task
+    class A:
+        executable = "ls"
+
+        class Inputs:
+            directory: os.PathLike = shell_arg(
+                help_string="input directory", argstr="", position=-1
+            )
+
+        class Outputs:
+            entries: list = shell_out(
+                help_string="list of entries returned by ls command",
+                callable=list_entries,
+            )
+
+    @shell_task
+    class B(A):
+        class Inputs(A.Inputs):
+            hidden: bool = shell_arg(
+                help_string="show hidden files",
+                argstr="-a",
+                default=False,
+            )
+
+    Path.touch(tmpdir / ".hidden")
+
+    b = B(directory=tmpdir, hidden=True)
+
+    assert b.inputs.directory == tmpdir
+    assert b.inputs.hidden
+
+    result = b()
+    assert result.output.entries == [".", "..", ".hidden"]
+
+
+def test_shell_missing_executable_static():
+    with pytest.raises(RuntimeError, match="should contain an `executable`"):
+
+        @shell_task
+        class A:
+            class Inputs:
+                directory: os.PathLike = shell_arg(
+                    help_string="input directory", argstr="", position=-1
+                )
+
+            class Outputs:
+                entries: list = shell_out(
+                    help_string="list of entries returned by ls command",
+                    callable=list_entries,
+                )
+
+
+def test_shell_missing_executable_dynamic():
+    with pytest.raises(RuntimeError, match="should contain an `executable`"):
+        A = shell_task(
+            "A",
+            executable=None,
+            input_fields={
+                "directory": {
+                    "type": os.PathLike,
+                    "help_string": "input directory",
+                    "argstr": "",
+                    "position": -1,
+                }
+            },
+            output_fields={
+                "entries": {
+                    "type": list,
+                    "help_string": "list of entries returned by ls command",
+                    "callable": list_entries,
+                }
+            },
+        )
+
+
+def test_shell_missing_inputs_static():
+    with pytest.raises(RuntimeError, match="should contain an `Inputs`"):
+
+        @shell_task
+        class A:
+            executable = "ls"
+
+            class Outputs:
+                entries: list = shell_out(
+                    help_string="list of entries returned by ls command",
+                    callable=list_entries,
+                )
+
+
+def test_shell_decorator_misuse(A):
+    with pytest.raises(
+        RuntimeError, match=("`shell_task` should not be provided any other arguments")
+    ):
+        shell_task(A, executable="cp")
```

Comment:
it might be an answer from one of my previous question. Is this a misuse because you try to overwrite `executable`?
Would it rise an error if `args` provided?
Perhaps the error message could be less general?